rivershared/util/dbutil/db_util.go

```diff
@@ -3,12 +3,44 @@ package dbutil
 import (
 	"context"
 	"fmt"
+	"time"
 
 	"github.com/riverqueue/river/riverdriver"
 )
 
+// RollbackWithoutCancel initiates a rollback, but one in which context is
+// overridden with context.WithoutCancel so that the rollback can proceed even
+// if a previous operation was cancelled. This decreases the chance that a
+// transaction is accidentally left open and in an ambiguous state.
+//
+// A rollback error is returned the same way most driver rollbacks return an
+// error, but given this is normally expected to be used in a defer statement,
+// it's unusual for the error to be handled.
+func RollbackWithoutCancel[TExec riverdriver.ExecutorTx](ctx context.Context, execTx TExec) error {
+	ctxWithoutCancel := context.WithoutCancel(ctx)
+
+	ctx, cancel := context.WithTimeout(ctxWithoutCancel, 5*time.Second)
+	defer cancel()
```

Comment:
super nit but I might prefer to inline this as it tends to be what I see in Go code in the wild, up to you though

```suggestion
	ctx, cancel := context.WithTimeout(context.WithoutCancel(ctx), 5*time.Second)
	defer cancel()
```